frontend/src/components/SeedlotRegistrationSteps/CollectionStep/index.tsx

```diff
@@ -107,20 +121,50 @@ const CollectionStep = (
     }
   };
 
-  const collectorNameInputRef = useRef<HTMLInputElement>(null);
-  const locationCodeInputRef = useRef<HTMLInputElement>(null);
-  const numberOfContainersInputRef = useRef<HTMLInputElement>(null);
-  const volumePerContainersInputRef = useRef<HTMLInputElement>(null);
-  const volumeOfConesInputRef = useRef<HTMLInputElement>(null);
-  const commentsInputRef = useRef<HTMLInputElement>(null);
-
-  const [isChecked, setIsChecked] = useState<boolean>(true);
-
-  const collectorAgencyisChecked = (event: React.ChangeEvent<HTMLInputElement>) => {
+  const collectorAgencyIsChecked = (event: React.ChangeEvent<HTMLInputElement>) => {
     const { checked } = event.target;
     setIsChecked(checked);
-    if (checked) {
-      handleFormInput('collectorAgency', defaultAgency, 'locationCode', defaultCode);
+    handleFormInput(
+      fieldsConfig.collector.name as keyof CollectionForm,
+      defaultAgency,
+      fieldsConfig.code.name as keyof CollectionForm,
+      defaultCode
+    );
+  };
+
+  const collectionVolumeInformationHandler = <K extends keyof CollectionForm>(
+    name: K,
+    value: CollectionForm[K]
+  ) => {
+    const numberOfContainers = +refControl.current[fieldsConfig.numberOfContainers.name].value;
+    const volumePerContainers = +refControl.current[fieldsConfig.volumePerContainers.name].value;
+    const volumeOfCones = +refControl.current[fieldsConfig.volumeOfCones.name].value;
```

Comment:
Can't this information come directly from the `state` prop?

Reply:
Tried it before but had a conflict with the state.